src/main/java/commands/AddDeadlineCommand.java

```diff
@@ -0,0 +1,59 @@
+package commands;
+
+import skibidi.Command;
+import skibidi.SkibidiException;
+import skibidi.Ui;
+import storage.TaskStorage;
+import storage.Deadline;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a command to add a deadline task.
+ */
+public class AddDeadlineCommand extends Command {
+    private final String description;
+    private final String deadline;
+
+    /**
+     * Creates a new AddDeadlineCommand.
+     *
+     * @param input The input string containing the deadline description and deadline date.
+     * @throws SkibidiException If the input string is in an invalid format.
+     */
+    public AddDeadlineCommand(String input) throws SkibidiException{
+        String[] parts = input.split("/by ");
+        if (parts.length < 2) {
+            throw new SkibidiException("Invalid deadline format. Usage: deadline [description] /by [date]");
```

Comment:
I like how you've given the user the expected input format in the error message!